### PR TITLE
Support ChainerX in CIFAR example

### PR DESCRIPTION
**Pull request by asi1024**
Part of #6661

### CODE REVIEW

File: examples/cifar/train_cifar_custom_loop.py [hunk 1]
```diff
@@ -102,9 +109,7 @@ def main():
             optimizer.lr *= 0.5
             print('Reducing learning rate to: {}'.format(optimizer.lr))
 
-        x_array, t_array = convert.concat_examples(batch, args.gpu)
-        x = chainer.Variable(x_array)
-        t = chainer.Variable(t_array)
```
Comment (niboshi) [inline examples/cifar/train_cifar_custom_loop.py]: Just to be clear, this is just a nonessential code improvement, right?

Comment (asi1024) [reply examples/cifar/train_cifar_custom_loop.py]: Yes. (Actually, I thought that it is necessary changes, but I found it unnecessary after I send this PR.)

File: examples/cifar/README.md [hunk 2]
```diff
@@ -22,5 +22,7 @@ train_cifar.py
 
 to run the CIFAR-100 dataset on GPU 1:
 ```
-train_cifar.py --gpu=1 --dataset='cifar100'
+train_cifar.py --device=1 --dataset='cifar100'
 ```
+
+TODO: Fix it.
```
[the remark this answers is not in the document]
Comment (asi1024) [reply examples/cifar/README.md]: Oops, I remove this line.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**